tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraFilterGetMethodContract.scala

```diff
@@ -518,4 +548,90 @@ trait LinagoraFilterGetMethodContract {
          |}""".stripMargin)
   }
 
+  @Test
+  def filterGetShouldReturnRuleOfSomeConditionsWhenUserHaveRuleOfSomeConditions(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[JmapGuiceCustomProbe])
+      .setRulesForUser(generateUsername(),
+        Rule.builder
+          .id(Rule.Id.of("1"))
+          .name("My first rule")
+          .conditionGroup(Rule.ConditionGroup.of(Rule.ConditionCombiner.OR, Rule.Condition.of(Rule.Condition.Field.SUBJECT, Rule.Condition.Comparator.CONTAINS, "question"),
+            Rule.Condition.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.CONTAINS, "user2")))
+          .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(generateMailboxIdForUser())))
+          .build)
+
+    val request =
+      s"""{
+         |  "using": ["com:linagora:params:jmap:filter" ],
+         |  "methodCalls": [
+         |    [
+         |      "Filter/get",
+         |        {
+         |          "accountId": "$generateAccountIdAsString",
+         |          "ids": ["singleton"]
+         |        },
+         |          "c1"]
+         |    ]
+         |}""".stripMargin
+
+    val response = `given`()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+      .when()
+      .post()
+      .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |  "sessionState": "${SESSION_STATE.value}",
+         |  "methodResponses": [[
+         |    "Filter/get", {
+         |      "accountId": "$generateAccountIdAsString",
+         |      "state": "0",
+         |      "list": [
+         |      {
+         |        "id": "singleton",
+         |        "rules": [
+         |          {
+         |            "name": "My first rule",
+         |            "conditionGroup": {
+         |              "conditionCombiner": "OR",
+         |              "conditions": [
+         |                {
```

Comment:
sorry I do not understand your idea

Reply:
{"conditions": [A, B] } != {"conditions": [B, A]}
```
    assertThatJson(response)
      .withOptions(new Options(Option.IGNORING_ARRAY_ORDER))
....
```

Reply:
if I add this, it will affect other arrays as well. So if it is not really necessary, I think we should not.

Reply:
in this HTTP response, I don't see another array that has > 1 element, 
I got a lot of ci unstable in the past because the order of elements is changed (distributed test)
if you are sure the order of the array will not be changed that ok